docs/configuration/indexes.md

```diff
@@ -0,0 +1,114 @@
+# Package indexes
+
+By default, uv uses the [Python Package Index (PyPI)](https://pypi.org) for dependency resolution
+and package installation. However, uv can be configured to use other package indexes, including
+private indexes, via the `[[tool.uv.index]]` configuration option (and `--index`, its analogous
+command-line option).
+
+## Defining an index
+
+To include an additional index when resolving dependencies, add a `[[tool.uv.index]]` entry to your
+`pyproject.toml`:
+
+```toml
+[[tool.uv.index]]
+# Optional, explicit name for the index.
+name = "pytorch"
+# Required URL for the index. Expects a repository compliant with PEP 503 (the simple repository API).
+url = "https://download.pytorch.org/whl/cpu"
+```
+
+Indexes are prioritized in the order in which they’re defined, such that the first index listed in
+the configuration file is the first index consulted when resolving dependencies, with indexes
+provided via the command line taking precedence over those in the configuration file.
+
+By default, uv includes the Python Package Index (PyPI) as the "default" index, i.e., the index used
+when a package is not found on any other index. To exclude PyPI from the list of indexes, set
+`default = true` on another index entry (or use the `--default-index` command-line option):
+
+```toml
+[[tool.uv.index]]
+name = "pytorch"
+url = "https://download.pytorch.org/whl/cpu"
+default = true
+```
```

Comment:
Interesting thing missing here. What if I want to disable having a default index entirely?

Reply:
`--no-index` does that, but otherwise there's no such thing.

Reply:
I think it'll be fairly common to want to disable PyPI entirely?

Reply:
That’s the same as defining a different default index.

Reply:
Yes, but it seems weird to need to do that. I might not want to mark my other index as a default for some reason? I don't fully understand the trade-offs. We can wait and see if it comes up, it doesn't block this pull request.

Reply:
Yeah, I think what you're getting at here is that `default = true` does two things: (1) replaces PyPI, (2) changes priority.